include/ignition/math/Plane.hh

```diff
@@ -119,6 +122,26 @@ namespace ignition
         return this->normal.Dot(_point) - this->d;
       }
 
+      /// \brief Get the intersection of an infinite line with the plane,
+      /// given the line's gradient and a point in parametrized space.
+      /// \param[in] _point A point that lies on the line.
+      /// \param[in] _gradient The gradient of the line.
+      /// \return The point of intersection. std::nullopt if the line is
+      /// parrallel to the plane.
+      public: std::optional<Vector3<T>> Intersection(
+        const Vector3<T> &_point,
+        const Vector3<T> &_gradient) const
+      {
+        if (abs(this->Normal().Dot(_gradient)) < 1e-6)
```

Comment:
I would recommend adding a `_tolerance` parameter that defaults to `1e-6` (see some of the APIs in [MassMatrix3.hh](https://github.com/ignitionrobotics/ign-math/blob/b54dce578d15ec24f2008b95d7b9d08f13ba5613/include/ignition/math/MassMatrix3.hh#L475) for similar examples)